File: csharp/MapleStory.OpenAPI/src/dto/CubeResultOptionDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.src.dto
{
    /// <summary>
    /// 큐브 결과 옵션 정보
    /// </summary>
    public class CubeResultOptionDTO
    {
        /// <summary>
        /// 옵션 이름
        /// </summary>
        [JsonProperty("value")]
        public string Value { get; set; }

        /// <summary>
        /// 옵션 등급
        /// </summary>
        [JsonProperty("grade")]
        public string Grade { get; set; }

        public PotentialOptionGrade GradeEnum
        {
            get {
                return Utils.PotentialOptionGradeFromString(this.Grade);
            }
        }
    }
}

File: js/src/maplestory/api/dto/history/cubeResultOptionDto.ts
import { PotentialOptionGrade, potentialOptionGradeFromString } from '@src/maplestory/api/dto/potentialOptionGrade';
import { CubeResultOptionDtoBody } from '@src/maplestory/api/response/history/cubeResultOptionDtoBody';

/**
 * 큐브 결과 옵션 정보
 */
class CubeResultOptionDto {
  /**
   * 옵션 명
   */
  value: string;

  /**
   * 옵션 등급
   */
  grade: string;

  constructor(obj: CubeResultOptionDtoBody) {
    const { value, grade } = obj;

    this.value = value;
    this.grade = grade;
  }

  get gradeEnum(): PotentialOptionGrade {
    return potentialOptionGradeFromString(this.grade);
  }
}

export { CubeResultOptionDto };
